packages/gpt/ad-manager.js

```
@@ -0,0 +1,122 @@
+import series from "run-series";
+import parallel from "run-parallel";
+
+import { getSlotConfig } from "./generate-config";
+import gptManager from "./gpt-manager";
+import pbjsManager from "./pbjs-manager";
+
+function AdManager(options = {}) {
+  if (!(this instanceof AdManager)) {
+    return new AdManager(options);
+  }
+
+  this.adQueue = [];
+  this.adUnit = options.adUnit;
+  this.networkId = options.networkId;
+  this.section = options.section;
+  this.initialised = false;
+}
+
+AdManager.prototype.isReady = function isReady() {
+  return this.initialised;
+};
+
+AdManager.prototype.init = function init(callback) {
+  // Load scripts if needed be
+  gptManager.loadScript();
+  pbjsManager.loadScript();
+
+  series(
+    [
+      // Set services config
+      parallel.bind(parallel, [
+        gptManager.setConfig.bind(gptManager),
+        pbjsManager.setConfig.bind(pbjsManager)
+      ]),
+      // Init services
+      parallel.bind(parallel, [
+        gptManager.init.bind(gptManager),
+        pbjsManager.init.bind(pbjsManager, this.adQueue)
+      ])
+    ],
+    err => {
+      if (err) throw new Error(err);
```

Are we going to have New Relic or some other tool? We should throw the error anyway, I'm not sure yet what's the guidelines for this particular project regarding that.